services/elasticache/pkg/resource/replication_group/custom_update_api.go

```diff
@@ -110,144 +180,184 @@ func (rm *resourceManager) updateReplicaCount(
+	res := &svcsdk.IncreaseReplicaCountInput{}
+	desiredSpec := desired.ko.Spec
 
 	res.SetApplyImmediately(true)
-	if desired.ko.Spec.ReplicationGroupID != nil {
-		res.SetReplicationGroupId(*desired.ko.Spec.ReplicationGroupID)
+	if desiredSpec.ReplicationGroupID != nil {
+		res.SetReplicationGroupId(*desiredSpec.ReplicationGroupID)
 	}
-	if desired.ko.Spec.NumNodeGroups != nil {
-		res.SetNodeGroupCount(*desired.ko.Spec.NumNodeGroups)
+	if desiredSpec.ReplicasPerNodeGroup != nil {
+		res.SetNewReplicaCount(*desiredSpec.ReplicasPerNodeGroup)
 	}
 
-	nodegroupsToRetain := []*string{}
-
-	// TODO: 	optional -only if- NumNodeGroups increases shards
-	// 			refer 'rl' to find out
-	if desired.ko.Spec.NodeGroupConfiguration != nil {
-		f13 := []*svcsdk.ReshardingConfiguration{}
-		for _, f13iter := range desired.ko.Spec.NodeGroupConfiguration {
-			f13elem := &svcsdk.ReshardingConfiguration{}
-			if f13iter.NodeGroupID != nil {
-				f13elem.SetNodeGroupId(*f13iter.NodeGroupID)
-				nodegroupsToRetain = append(nodegroupsToRetain, &(*f13iter.NodeGroupID))
+	if desiredSpec.NodeGroupConfiguration != nil {
+		shardsConfig := []*svcsdk.ConfigureShard{}
+		for _, desiredShard := range desiredSpec.NodeGroupConfiguration {
+			shardConfig := &svcsdk.ConfigureShard{}
+			if desiredShard.NodeGroupID != nil {
+				shardConfig.SetNodeGroupId(*desiredShard.NodeGroupID)
+			}
+			if desiredShard.ReplicaCount != nil {
+				shardConfig.SetNewReplicaCount(*desiredShard.ReplicaCount)
 			}
-			f13elemf2 := []*string{}
-			if f13iter.PrimaryAvailabilityZone != nil {
-				f13elemf2 = append(f13elemf2, &(*f13iter.PrimaryAvailabilityZone))
+			shardAZs := []*string{}
+			if desiredShard.PrimaryAvailabilityZone != nil {
+				shardAZs = append(shardAZs, desiredShard.PrimaryAvailabilityZone)
 			}
-			if f13iter.ReplicaAvailabilityZones != nil {
-				for _, f13elemf2iter := range f13iter.ReplicaAvailabilityZones {
-					var f13elemf2elem string
-					f13elemf2elem = *f13elemf2iter
-					f13elemf2 = append(f13elemf2, &f13elemf2elem)
+			if desiredShard.ReplicaAvailabilityZones != nil {
+				for _, desiredAZ := range desiredShard.ReplicaAvailabilityZones {
+					shardAZs = append(shardAZs, desiredAZ)
 				}
-				f13elem.SetPreferredAvailabilityZones(f13elemf2)
 			}
-			f13 = append(f13, f13elem)
+			if len(shardAZs) > 0 {
+				shardConfig.SetPreferredAvailabilityZones(shardAZs)
+			}
+			shardsConfig = append(shardsConfig, shardConfig)
 		}
-		res.SetReshardingConfiguration(f13)
+		res.SetReplicaConfiguration(shardsConfig)
 	}
 
-	// TODO: 	optional - only if -  NumNodeGroups decreases shards
-	//			refer 'rl' to find out
-	// res.SetNodeGroupsToRemove() or res.SetNodeGroupsToRetain()
-	res.SetNodeGroupsToRetain(nodegroupsToRetain)
-
 	return res, nil
 }
 
-// new(IncreaseReplicaCount)RequestPayload returns an SDK-specific struct for the HTTP request
+// new(DecreaseReplicaCount)RequestPayload returns an SDK-specific struct for the HTTP request
```

Comment:
Can you remove the parens around DecreaseReplicaCount above?

Reply:
Updated source file.